mali_counter.cpp

```diff
@@ -443,23 +439,30 @@ const uint32_t *MaliCounter::get_counters() const
 	return _raw_counter_buffer.data();
 }
 
-const uint32_t *MaliCounter::get_counters(mali_userspace::MaliCounterBlockName block, int core) const
+const uint32_t *MaliCounter::get_counters(mali_userspace::MaliCounterBlockName block, int index) const
 {
 	switch (block)
 	{
 		case mali_userspace::MALI_NAME_BLOCK_JM:
 			return _raw_counter_buffer.data() + mali_userspace::MALI_NAME_BLOCK_SIZE * 0;
 		case mali_userspace::MALI_NAME_BLOCK_MMU:
-			return _raw_counter_buffer.data() + mali_userspace::MALI_NAME_BLOCK_SIZE * 2;
+			if (index < 0 || index >= _num_l2_slices)
+			{
+				throw std::runtime_error("Invalid core number.");
```

Comment:
s/core/slice?

Reply:
Done in https://github.com/ARM-software/HWCPipe/pull/7/commits/12c14b6ad8acb9c2782d2e4b3b82711bb337b8ec